### ISSUE TITLE
a11y issues - contrast

### ISSUE THREAD

**Issue by ctcpip** [Ileriayo/markdown-badges#220]
There are a number of badges that lack adequate contrast, so don't meet a11y requirements such as WCAG AA.  Although end users can change the color values to whatever they see fit, most people will just use the defaults.  The default values should at least meet WCAG AA.  One example is the Java shield, which has a contrast of `2.53:1`.  `WCAG 2.0 level AA` requires a contrast ratio of at least `4.5:1`

**Comment by mehtaAnsh** [comment]
Hey @ctcpip , we do not have any guidelines to meet a11y requirements. If you do find any errors, you can create a Pull Request and I would gladly merge it. 

@Ileriayo can we think of any pre-commit hook/action that can check for these guidelines? It can be easily managed then.

**Comment by ctcpip** [comment]
FWIW, I did spend some time writing a parser to extract each fg and bg color and check the contrast ratio, and output any that did not meet the requirement.  However, due to inconsistencies and lack of uniformity, it became clear it would need to handle all these edge cases.

To move forward, I would have to first write a general validator for each entry, fix the anomalies by updating the readme, and then could run it through the contract check parser.  Ideally, both of these tasks would be part of either a pre-commit hook, or an action required to pass in order to merge.  (But only after fixing all the existing issues first, or it would never pass.)

**Comment by ctcpip** [comment]
It was easier just to write an automated way to do the whole thing.  Thanks for the inspiration!  Check out the `limitations` section though...  Nothing we can do about the text in many cases, until it's fixed on their end.

https://github.com/ctcpip/a11y-markdown-badges

**Comment by mehtaAnsh** [comment]
That's a great way to implement. We can have something similar like this here in the form of a pre-commit hook.

**Comment by ctcpip** [comment]
@mehtaAnsh - I've been busy... check out https://github.com/a11y-badges/a11y-badges -- and [my readme](https://github.com/ctcpip) for a full demo.  thanks again for the inspiration!

**Comment by mehtaAnsh** [comment]
The implementation looks great. Can't think of any way to integrate this with this repository as of now. Great work!

**Comment by ctcpip** [comment]
closing in favor of [a11y-markdown-badges](https://github.com/a11y-badges/a11y-markdown-badges) / [a11y-badges](https://github.com/a11y-badges/a11y-badges)
